crazyflow/sim/visualize.py

```
@@ -69,6 +69,61 @@ def draw_points(sim: Sim, points: NDArray, rgba: NDArray | None = None, size: fl
         )
 
 
+def change_material(
+    sim: Sim,
+    mat_name: str,
+    drone_ids: NDArray,
+    rgba: NDArray | None = None,
+    emission: NDArray | None = None,
+):
+    """Change the material of all drones matching the mask.
+
+    Args:
+        sim: The simulation.
+        mat_name: The name of the material to change.
+        drone_ids: Array of drone indices to modify, shape (n,), dtype=int.
+        rgba: The RGBA color to set, should be of shape (n, 4) or (4,) to be auto-broadcasted.
+        emission: The emission value of material, should be of shape (n,) or scalar.
+    """
+    if drone_ids.ndim != 1:
+        raise ValueError(f"drone_ids must be 1D array, got shape {drone_ids.shape}")
+    if np.any(drone_ids < 0) or np.any(drone_ids >= sim.n_drones):
+        raise ValueError(f"drone_ids must be in range [0, {sim.n_drones - 1}], got {drone_ids}")
+
+    if rgba is not None:
+        rgba = np.asarray(rgba, dtype=float)
+        try:
+            # this returns itself if rgba is already the right shape
+            rgba = np.broadcast_to(rgba, (len(drone_ids), 4)).copy()
+        except Exception:
+            raise ValueError(f"rgba must be shape (4,) or ({len(drone_ids)}, 4), got {rgba.shape}")
```

Why the try block? If this fails, just let it fail. Also, try to avoid bare Exceptions at all cost. They can be extremely confusing for users. Anything could go wrong, but they wouldn't see any of it because the bare except just swallows the error.

---

Removed. 
The np.broadcast_to() will report:
"ValueError: operands could not be broadcast together with remapped shapes [original->remapped]: (3,4)  and requested shape (13,4)"
if shape is incorrect, which is also clear.